lib/transform-stream.js

```diff
@@ -40,23 +72,35 @@ class TransformStreamSource {
 
     // eslint-disable-next-line no-unmodified-loop-condition
     while (!enqueued) {
-      const data = await this.reader.read()
-      if (data.done) {
-        if (this.transformer.flush) {
-          await this.transformer.flush(controller)
+      try {
+        const data = await this.reader.read()
+        if (data.done) {
+          if (this.transformer?.flush) {
+            await this.transformer.flush(controller)
+          }
+          controller.close()
+          this.doneCb(null)
+          return
         }
-        controller.close()
-        return
-      }
-      if (this.transformer.transform) {
-        await this.transformer.transform(data.value, wrappedController)
-      } else {
-        wrappedController.enqueue(data.value)
+        if (this.transformer?.transform) {
+          await this.transformer.transform(data.value, wrappedController)
+        } else {
+          wrappedController.enqueue(data.value)
+        }
+      } catch (err) {
+        this.doneCb(err)
+        throw err
       }
     }
   }
 
   async cancel (reason) {
-    return await this.reader.cancel(reason)
+    await this.reader.cancel(reason)
+    if (reason instanceof Error) {
+      this.doneCb(reason)
+    } else {
+      this.doneCb(new Error(`stream cancelled; reason: ${reason}`))
+    }
+    return reason
```

Comment:
I think you can remove the `return` here.

Reply:
Good point!